### PR TITLE
feat(connector-insights): track manifest usage of $parameters and custom component classes

### PR DESCRIPTION
**Pull request by natikgadzhi**
## What

Adds `manifest_uses_parameters`, `manifest_uses_custom_components`, and `manifest_custom_component_classes` to `connector_insights`.

With this information, we will have more data in Migration Tracker dashboard on how to approach manifest-only migration.

## Context

- There are 219 manifests with $parameters . This is going to be a blocker, and we should resolve them.
- `find . -name "manifest.yaml" -type f | xargs grep -l "class_name:" > manifests-with-custom-components.txt`
  - There are 67 manifests with custom components total.

## User Impact

None, dev tool change.

## Next steps

Once this change lands, we need to regenerate insights, and then [bring them to airbyte-warehouse](https://github.com/airbytehq/airbyte-warehouse/issues/2431).

### CODE REVIEW

**Review by girarda — APPROVED** [review]
🚢

File: airbyte-ci/connectors/connectors_insights/src/connectors_insights/insights.py [hunk 1]
```diff
@@ -23,6 +23,19 @@
     from connector_ops.utils import Connector  # type: ignore
 
 
+def get_manifest_inferred_insights(connector: Connector) -> dict:
+    manifest = connector.manifest_path.read_text()
```
Comment (alafanechere) [inline airbyte-ci/connectors/connectors_insights/src/connectors_insights/insights.py:27]: @natikgadzhi this raises a `FileNotFound`error on connectors without manifests.

**Pull request merged**